app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,148 @@ def registry_authorize
+        segments[2] == "id" && (segments[6] == "blobs" || segments[6] == "manifests")
+
+        return {
+          valid_format: true,
+          schema: "id",
+          organization: segments[3],
+          product: segments[4],
+          name: segments[5]
+        }
+      else
+        return {valid_format: false}
+      end      
+    end
+
+    def check_blob_push_field_syntax(props)
+      # check basic url field syntax
+      unless props[:valid_format]
+        render_error('custom_error', status: unprocessable_entity,
+          locals: {message: "Invalid format. Container pushes should follow '/organization_label/product_label/name' OR '/organization_id/product_id/name' schema." })
+        return false
+      end
+      return true
+    end
+
+    def check_blob_push_org(props)
+      # find the org using the appropriate schema
+      if props[:schema] == "label"
+        org_label = props[:organization]
+        unless org_label.present? && org_label.length > 0
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Organization label cannot be blank." })
+          return false
+        end
+        org = Organization.where("LOWER(label) = '#{org_label}'") # convert to lowercase
+        if org.length > 1 # reject ambiguous orgs (possible due to lowercase conversion)
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Organization label is ambiguous" })
+          return false
+        elsif org.length == 0
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Organization not found: '#{org_label}'" })
+          return false
+        end
+        @organization = org.first
+      else
+        org_id = props[:organization]
+        unless org_id.present? && org_id == org_id.to_i.to_s # checks if id string is integer and has no leading zeros
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Organization id must be an integer without leading zeros." })
+          return false
+        end
+        @organization = Organization.find(org_id.to_i)
+      end
+      return true
+    end
+
+    def check_blob_push_product(props)
+      # find the product using the appropriate schema
+      if props[:schema] == "label"
+        prod_label = props[:product]
+        unless prod_label.present? && prod_label.length > 0
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Product label cannot be blank." })
+          return false
+        end
+        product = @organization.products.where("LOWER(label) = '#{prod_label}'") # convert to lowercase
+        if product.length > 1 # reject ambiguous products (possible due to lowercase conversion)
+          render_error('custom_error', status: unprocessable_entity,
```

Comment:
Like below, the error here reaches the user as a 500 and the message is not carried along. If you start using the OCI error spec, you'll be able to get your error message across.

Reply:
Error handling is now using OCI spec, thank you.